exercises/practice/clock/.meta/example.nim

```diff
@@ -1,18 +1,31 @@
-import strformat
+import std/[math, strformat]
 
-type Clock = int
-type Time = tuple[h, m: int]
+# `euclMod` was added to `std/math` in Nim 1.6.0.
+# Define `euclMod` ourselves when we are compiling with an earlier Nim version.
+when not compiles(euclMod(3, 2)):
+  func euclMod*[T: SomeNumber](x, y: T): T =
+    ## Returns euclidean modulo of `x` by `y` (returning a non-negative number).
+    result = x mod y
+    if result < 0:
+      result += abs(y)
 
-func create*(t: Time): Clock =
-  result = (t.h * 60 + t.m) mod 1440
-  if result < 0:
-    result += 1440
+type
+  Clock* = object
+    hour*: range[0..23]
+    minute*: range[0..59]
 
-func toStr*(c: Clock): string =
-  &"{c div 60:02}:{c mod 60:02}"
+  Minutes* = distinct int
 
-func add*(t: Time, v: int): int =
-  create (t.h, t.m + v)
+func initClock*(hour, minute: int): Clock =
+  const minutesInHour = 60
+  let numMinutes = (hour * minutesInHour + minute).euclMod(minutesInHour * 24)
+  Clock(hour: numMinutes div minutesInHour, minute: numMinutes mod minutesInHour)
```

Comment:
My personal (non-blocking) preference is to use `result` here as is recommended as nim style in NEP-1

Reply:
I agree for short initialization procs, but this is multiple lines